cg/meta/upload/mutacc.py

```diff
@@ -0,0 +1,194 @@
+            return {'case': mutacc_case, 'causatives': mutacc_variants}
+        return {}
+
+    def extract_reads(self, case: dict):
+        """Use mutacc API to extract reads from case"""
+        data = self._data(case)
+        if data:
+            LOG.info("Extracting reads from case %s", case['_id'])
+            self.mutacc_auto.extract_reads(case=data['case'], variants=data['causatives'])
+
+    def import_cases(self):
+        """Use mutacc API to import cases to database"""
+        LOG.info('importing cases into mutacc database')
+        self.mutacc_auto.import_reads()
+
+    @staticmethod
+    def _has_bam(case: dict) -> bool:
+
+        """
+            Check that all samples in case has a given path to a bam file,
+            and that the file exists
+
+            Args:
+                case (dict): case dictionary from scout
+
+            Returns:
+                (bool): True if all samples has valid paths to a bam-file
+
+        """
+
+
+        for sample in case['individuals']:
+
+            if sample.get('bam_file', None) is None:
+                LOG.warning("sample %s in case %s is missing bam fille",
+                            sample['individual_id'], case['_id'])
+                return False
+
+            elif not os.path.isfile(sample['bam_file']):
+                LOG.warning("sample %s in %s has non existing bam_file",
+                            sample['individual_id'], case['_id'])
+                return False
+
+        return True
+
+    @staticmethod
+    def _has_causatives(case: dict) -> bool:
+        """
+            Check that the case has marked causative variants in scout
+
+            Args:
+                case (dict): case dictionary from scout
+
+            Returns:
+                (bool): True if case has marked causative variants in scout
+        """
+        if case.get('causatives'):
+            return True
+
+        LOG.warning("case %s has no marked causatives in scout", case['_id'])
+        return False
+
+
+# Reformat scout noutput to mutacc input
+
+MAPPER = namedtuple('mapper', ['field_name_1', 'field_name_2', 'conv'])
+
+def remap(input_dict: dict, mapper_list: list) -> dict:
+    """
+        Reformat dict from one application to be used by another
+
+        Args:
+            input_dict (dict): dictionary to be converted
+            mapper_list (list(MAPPER)): list of mapper objects
+
+        Returns:
+            output_dict (dict): conveted dictionary
+
+    """
+    output_dict = {}
+    for field in mapper_list:
+        if input_dict.get(field.field_name_1, None) is not None:
+            output_dict[field.field_name_2] = field.conv(input_dict[field.field_name_1])
+    return output_dict
+
+
+SCOUT_TO_MUTACC_SAMPLE = (
+    MAPPER('individual_id', 'sample_id', str),
+    MAPPER('sex', 'sex',
+           lambda sex: 'male' if sex == '1' else 'female' if sex == '2' else 'unknown'),
```

Comment:
does this work as intended? a bit hard for me to read..

Reply:
:+1: Exactly, please don't stack ternary operators. Move them proper if statements in its own function :)

Reply:
Sure it works, but I agree on the readability part! will change this.